### PR TITLE
Removed end-of-stream, finally as fast as morgan.

### PR DESCRIPTION
**Pull request by mcollina**
@davidmarkclements 🎉 !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,9 @@ An [express](http://npm.im/express) middleware to log with
 From our benchmarks, this is 4x faster than
 [express-bunyan-logger](http://npm.im/express-bunyan-logger) and 25%
 faster than [winston-express](http://npm.im/winston-express).
-It is slightly slower (< 5%) than [morgan](http://npm.im/morgan),
-which is a usecase-specific logger. With
+It is as fast as [morgan](http://npm.im/morgan),
+which is a usecase-specific logger, i.e. it does not provide a
+request-specific logger. With
```
Comment (davidmarkclements) [inline README.md]: also - morgan doesn't do JSON right?

**Comment by davidmarkclements** [comment]
chatting with @mafintosh - we may need to check for on `.('close'` event,
way to test this is to forge an HTTP req (e.g. via telnet) that doesn't end

**Comment by davidmarkclements** [comment]
also we need to a check that handlers aren't called twice

**Comment by mcollina** [comment]
`req.on('close')` or `res.on('close')`?

We are currently avoid double calls by removing our listener: https://github.com/mcollina/express-pino-logger/blob/master/logger.js#L28-L29

**Comment by davidmarkclements** [comment]
ah yes - I wonder if it's faster to have a flag check? back in the jQuery days it was always faster to flag check than to toggle listeners

**Comment by mcollina** [comment]
What about `req` or `res`? Can you add a unit test for this case, I'm missing why that could happen.
Regarding the flag check, we still need to remove the handler from `'error'`.

**Pull request merged**